### PR TITLE
fixing 94X scale & smearing and updates

### PR DESCRIPTION
**Pull request by Sam-Harper**
Hi Marco,

So this fixes the 2017 94X  electrons scale and smearing. We have to rerun the scale and smearing due to a long standing bug in it.  Not necessary for 2016.

So I changed the names to reflect this, made the scale and smearing file explicit just incase we changed it ever. I also snuck in a little fix, the HEEP ID actually uses a different track isolation. So I stuck that into the nanoAOD. Its a variable that is often zero and only for 35 GeV electrons so hopefully the impact is minimal.

I have some fixes for 2016 to come but as 2017 is leaving first important to get that inplace. 

Best,
Sam

### CODE REVIEW

File: PhysicsTools/NanoAOD/python/electrons_cff.py [hunk 1]
```diff
@@ -202,10 +204,20 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
     mvaFall17V2noIso = None,
 )
 run2_nanoAOD_94XMiniAODv1.toModify(slimmedElectronsWithUserData.userFloats,
-    ecalTrkEnergyErrPostCorr = cms.InputTag("calibratedPatElectrons94Xv1","ecalTrkEnergyErrPostCorr"),
-    ecalTrkEnergyPreCorr     = cms.InputTag("calibratedPatElectrons94Xv1","ecalTrkEnergyPreCorr"),
-    ecalTrkEnergyPostCorr    = cms.InputTag("calibratedPatElectrons94Xv1","ecalTrkEnergyPostCorr"),
+    ecalTrkEnergyErrPostCorr = cms.InputTag("calibratedPatElectrons94X","ecalTrkEnergyErrPostCorr"),
```
Comment (peruzzim) [inline PhysicsTools/NanoAOD/python/electrons_cff.py]: I would find it cleaner to just append the New suffix also for v1, as in any case this is just an internal variable, to reduce duplication and customization here and later on

File: PhysicsTools/NanoAOD/python/electrons_cff.py [hunk 2]
```diff
@@ -327,6 +339,7 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
         pfRelIso03_all = Var("userFloat('PFIsoAll')/pt",float,doc="PF relative isolation dR=0.3, total (with rho*EA PU corrections)"),
         jetRelIso = Var("?userCand('jetForLepJetVar').isNonnull()?(1./userFloat('ptRatio'))-1.:userFloat('PFIsoAll04')/pt",float,doc="Relative isolation in matched jet (1/ptRatio-1, pfRelIso04_all if no matched jet)",precision=8),
         dr03TkSumPt = Var("?pt>35?dr03TkSumPt():0",float,doc="Non-PF track isolation within a delta R cone of 0.3 with electron pt > 35 GeV",precision=8),
+        dr03TkSumPtHEEP = Var("?pt>35?userFloat('dr03TkSumPtHEEP'):0",float,doc="Non-PF track isolation within a delta R cone of 0.3 with electron pt > 35 GeV used in HEEP ID",precision=8),
```
Comment (peruzzim) [inline PhysicsTools/NanoAOD/python/electrons_cff.py:343]: in what does this differ from the variable above (dr03TkSumPt)?

**Review by peruzzim — CHANGES_REQUESTED** [review]

File: PhysicsTools/NanoAOD/python/electrons_cff.py (hunk 2 above)
Comment (Sam-Harper) [reply PhysicsTools/NanoAOD/python/electrons_cff.py:343]: it has different veto settings and has significantly more sig/bkg rejection.

File: PhysicsTools/NanoAOD/python/electrons_cff.py (hunk 1 above)
Comment (Sam-Harper) [reply PhysicsTools/NanoAOD/python/electrons_cff.py]: either way works, I have no opinion here, I'll make the change.

**Review by peruzzim — APPROVED** [review]

**Pull request merged**